src/AzureCacheForRedis.cs

```diff
@@ -113,7 +113,7 @@ public static async Task<ConfigurationOptions> ConfigureForAzureWithTokenCredent
         }
 
         configurationOptions.Defaults = optionsProvider;
-        configurationOptions.User ??= azureCacheOptions.GetUserName(configurationOptions.Password);
+        optionsProvider._user = configurationOptions.User ?? azureCacheOptions.GetUserName(configurationOptions.Password);
```

Comment:
To confirm: does `configurationOptions.User ` still need to be updated?

Reply:
Hello. FYI, from here: https://github.com/StackExchange/StackExchange.Redis/blob/39b992fc4b7a32a6d96ff02f516f863807501a0c/src/StackExchange.Redis/ConfigurationOptions.cs#L535
The User value defaults to Defaults.User if not set. Maybe you can even just do:
        optionsProvider._user = azureCacheOptions.GetUserName(configurationOptions.Password);

Reply:
Correct, we don't need to set `configurationOptions.User` directly because it will fall through to `optionsProvider._user`. 

But it's still possible for a user to override the automatically extracted user name if they need to for some reason. They do that by setting `configurationOptions.User` to something else either before or after this initialization happens. That's why we call `GetUserName()` only if `configurationOptions.User` is null.